public/components/app.tsx

```diff
@@ -38,27 +51,49 @@ export const MlCommonsPluginApp = ({
   http,
   chrome,
   data,
+  dataSource,
+  dataSourceManagement,
+  savedObjects,
+  setActionMenu,
 }: MlCommonsPluginAppDeps) => {
+  const dataSourceEnabled = !!dataSource?.dataSourceEnabled;
```

Comment:
nit: we should be able to tell if the data source plugin is enabled via `!!dataSource`

Reply:
Do you mean we don't need to read the `dataSourceEnabled` property in `dataSource`?  The multi data source will always be enabled if `dataSource` is truly.